hasher-matcher-actioner/hmalib/common/timebucketizer.py

```diff
@@ -174,8 +176,61 @@ def get_records(
 
         for file in file_list:
             with open(file, "r") as my_file:
-                content_list.extend(
-                    list(map(type_class().from_csv, csv.reader(my_file)))
-                )
+                content_list.extend(list(map(type_class.from_csv, csv.reader(my_file))))
 
         return content_list
+
+    @staticmethod
+    def squash_directory(directory_path):
+        file_list = []
+        for file in os.listdir(directory_path):
+            file_path = os.path.join(directory_path, file)
+            if file.startswith("squash"):
+                logger.info("Directory (%s) already squashed.", directory_path)
+                return
+            elif os.path.isfile(file_path):
+                file_list.append(file_path)
+
+        with open(
+            os.path.join(directory_path, "squash_" + str(uuid.uuid1())) + ".csv",
+            "w",
+        ) as new_file:
+            for file in file_list:
+                logger.info("getting content of: %s", file)
+                with open(file, "r") as reader:
+                    new_file.write(reader.read())
+                logger.info("removing file: %s", file)
+                os.remove(file)
+        logger.info("successfully squashed: %s", directory_path)
+
+    @staticmethod
+    def squash_content(
+        type: str,
+        storage_path: str,
+        bucket_width: datetime.timedelta,
+        max_bucket_start_time: datetime.datetime,
```

Comment:
nit: I do not expect to pass a max value BEFORE a min value. Natural ordering suggests min and then max.